src/main/scala/com/databricks/labs/mosaic/expressions/geometry/ST_Envelope.scala

```diff
@@ -0,0 +1,85 @@
+package com.databricks.labs.mosaic.expressions.geometry
+
+import com.databricks.labs.mosaic.codegen.format.ConvertToCodeGen
+import org.apache.spark.sql.catalyst.expressions.UnaryExpression
+import org.apache.spark.sql.catalyst.expressions.NullIntolerant
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.types.DataType
+import com.databricks.labs.mosaic.core.geometry.api.GeometryAPI
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode}
+import org.apache.spark.sql.catalyst.expressions.ExpressionInfo
+
+case class ST_Envelope(inputGeom: Expression, geometryAPIName: String) extends UnaryExpression with NullIntolerant {
+
+    override def child: Expression = inputGeom
+
+    override def dataType: DataType = inputGeom.dataType
+
+    override protected def nullSafeEval(input: Any): Any = {
+        val geometryAPI = GeometryAPI(geometryAPIName)
+        val geometry = geometryAPI.geometry(input, inputGeom.dataType)
+        val envelope = geometry.envelope
+        geometryAPI.serialize(envelope, inputGeom.dataType)
+    }
+
+    override def makeCopy(newArgs: Array[AnyRef]): Expression = {
+        val asArray = newArgs.take(1).map(_.asInstanceOf[Expression])
+        val res = ST_Envelope(asArray.head, geometryAPIName)
+        res.copyTagsFrom(this)
+        res
+    }
+
+    override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode =
+        nullSafeCodeGen(
+          ctx,
+          ev,
+          leftEval => {
+              val geometryAPI = GeometryAPI.apply(geometryAPIName)
+              val envelope = ctx.freshName("envelope")
+              val geometryClass = geometryAPI.geometryClass
+              val (inCode, geomInRef) = ConvertToCodeGen.readGeometryCode(ctx, leftEval, inputGeom.dataType, geometryAPI)
+              val (outCode, outGeomRef) = ConvertToCodeGen.writeGeometryCode(ctx, envelope, inputGeom.dataType, geometryAPI)
+
+              geometryAPI.name match {
+                  case "ESRI" => geometryAPI.codeGenTryWrap(s"""
```

Comment:
@tdikland @edurdevic we should think about abstracting this to make the generation of expressions faster.

I would think something like ST_AttributeBase trait that extends UnaryExpression would work since it would wrap a lot of this logic. Basically the only unique part of this whole function are lines 46 and 52.
This is not needed for this PR but would like us to start designing this.
Same goes for ST_BinaryBase - again there is about 60% of the code that is repeatable.